### ISSUE TITLE
Memoize result for a given input

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/prompter#19]
A great optimization, I believe, would be to memoize the source `update` results depending on the input.

Example:

- Type "foo", takes 10s to compute.
- Type "bar", takes 15s to compute.
- Type "foo" again.

With the current `prompter`, the last "foo" would take another 10s.  With memoization, it would be instantaneous.

In terms of memory, I don't believe it's a big deal since prompters are short-lived objects.

**Comment by aartaka** [comment]
Yeah, boolean toggle is the best solution here. But I'm somewhat cautious about setting it to T by default—seems like a potential foot-shooting behavior. 

I mean, in case of Nyxt we can add a proxy source class with `memoize-p` set to T, but in the `prompter` itself I'd rather have it at NIL to not cause Heisenbugs downstream.

**Comment by Ambrevar** [comment]
Fair enough, it's super easy to subclass `prompter` anyways :)
